osidb/models.py

```diff
@@ -1855,30 +1856,41 @@ def _validate_sofware_collection(self):
                 "does not match any valid collection",
             )
 
-    def _validate_exceptional_affectedness_resolution(self):
+    def _validate_historical_affectedness_resolution(self):
         """
-        Alerts that an old flaw have empty affectedness.
-        (Only accepts WONTFIX and DEFER as resolution, validated by other validation)
+        Alerts that an old flaw has an affectedness/resolution combination that is now invalid,
+        but was valid in the past.
         """
-        valid_resolutions = [
-            Affect.AffectResolution.WONTFIX,
-            Affect.AffectResolution.DEFER,
-        ]
         if (
-            self.affectedness == Affect.AffectAffectedness.NOVALUE
-            and self.resolution in valid_resolutions
+            self.resolution
+            in AFFECTEDNESS_HISTORICAL_VALID_RESOLUTIONS[self.affectedness]
         ):
-            self.alert(
-                "flaw_exceptional_affect_status",
-                f"Affect ({self.uuid}) for {self.ps_module}/{self.ps_component} is in "
-                "a exceptional state having no affectedness.",
-            )
+            # Alert only if creating a new record or modifying the affectedness or resolution
+            # specifically into the invalid combination.
+            need_alert = self._state.adding
+            if not need_alert:
+                old_affect = Affect.objects.get(uuid=self.uuid)
+                need_alert = (
+                    self.affectedness != old_affect.affectedness
+                    or self.resolution != old_affect.resolution
+                )
+            if need_alert:
+                self.alert(
+                    "flaw_historical_affect_status",
+                    f"Affect ({self.uuid}) for {self.ps_module}/{self.ps_component} has an "
+                    "historical affectedness/resolution combination which is not valid anymore.",
+                )
```

Comment:
The logic should be slightly more restrictive. There should always be some kind of warning. In the case of `need_alert == False` we should only store the alert by `self.alert` but when `need_alert == True` we should really `raise ValidationError`. The idea is that we want to prevent any new values like this and `raise` will really interrupt the whole query processing while for the old data we still want to inform the user but softly.

Reply:
I modified it so that it raises an error when creating a new record instead, and throws an alert when modifying existing data